### PR TITLE
feat(DIA-105): Move follow button for artist and display count on button

### PR DESCRIPTION
**Pull request by jo-rs**
This PR resolves [DIA-105]

### Description

- Uses the scroll offset enhancement from the tabs to hide the follow button when the title is shown in the header
- Small text changes in the artist name, origin

https://github.com/artsy/eigen/assets/15792853/82152727-b2ca-4e3d-976d-731740f5b0e5

Note: there is some flickering with the image, I checked the performance in general but looks more related to the relay after the data update. We can create a follow-up ticket to investigate it further if necessary.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- moves follow button to the Artist header navigation bar - araujobarrert

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-105]: https://artsyproduct.atlassian.net/browse/DIA-105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9fe4f2ef821573c3def9976ef6a51758c12931d4
</p>

File: src/app/Components/Artist/useFollowArtist.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            : artistCount - 1,
+          "follows"
+        )
+      },
+      onCompleted: successfulFollowChange,
+      onError: failedFollowChange,
+    })
+  }
+
+  const successfulFollowChange = () => {
```
Comment (damassi) [inline src/app/Components/Artist/useFollowArtist.ts:81]: `onFollowChangeSuccess`

File: src/app/Components/Artist/useFollowArtist.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      action_type: Schema.ActionTypes.Success,
+      owner_id: data.internalID,
+      owner_slug: data.slug,
+      owner_type: Schema.OwnerEntityTypes.Artist,
+    })
+
+    setIsLoading(false)
+  }
+
+  const failedFollowChange = () => {
```
Comment (damassi) [inline src/app/Components/Artist/useFollowArtist.ts:98]: `onFollowChangeError`

File: src/app/Components/Artist/useFollowArtist.ts (hunk 1 above)
Comment (araujobarret) [reply src/app/Components/Artist/useFollowArtist.ts:81]: will 🧹 tomorrow, was some copy/pasta

**Review by damassi — DISMISSED** [review]
Updates look great @araujobarret

File: src/app/Scenes/Artist/Artist.tsx [hunk 3]
```diff
@@ -106,14 +111,23 @@ export const Artist: React.FC<ArtistProps> = (props) => {
...
           BelowTitleHeaderComponent={() => (
-            <ArtistHeaderFragmentContainer artist={artistAboveTheFold!} />
+            <ArtistHeaderFragmentContainer
+              artist={artistAboveTheFold!}
+              onLayout={({ nativeEvent }) => {
+                if (headerHeight !== nativeEvent.layout.height) {
+                  setHeaderHeight(nativeEvent.layout.height)
+                }
+              }}
+            />
```
Comment (MounirDhahri) [inline src/app/Scenes/Artist/Artist.tsx:130]: **issue(non-blocker):** following the artist causes this component to refresh and therefore the flashing you see in the screen. If you refactor this to not rely on react ability to tell what changed it should probably fix it. Something like
```typescript
//
BelowTitleHeaderComponent={renderBelowTitleHeaderComponent}
//
```

```typescript
  const renderBelowTitleHeaderComponent = useCallback(() => {
    return (
      <ArtistHeaderFragmentContainer
        artist={artistAboveTheFold!}
        onLayout={({ nativeEvent }) => {
          if (headerHeight !== nativeEvent.layout.height) {
            setHeaderHeight(nativeEvent.layout.height)
          }
        }}
      />
    )
  }, [artistAboveTheFold, headerHeight])
```

**Pull request merged**